docs/data/date-pickers/custom-field/custom-field.md

```diff
@@ -154,3 +154,309 @@ and you don't want the UI to look like a Text Field, you can replace the field w
+    const newValue = dayjs(newInputValue, format);
+    setInputValue(newInputValue);
+    onChange(newValue, { validationError: getValidationErrorForNewValue(newValue) });
+  };
+
+  return (
+    <TextField
+      value={inputValue}
+      onChange={handleChange}
+      error={hasValidationError}
+    />
+  );
+}
+```
+
+#### Localized placeholder
+
+You can use the `useParsedFormat` to get a clean placeholder.
+This hook applies two main transformation on the format:
+
+1. It replaces all the localized tokens (i.e: `L` for a date with `dayjs`) with there expanded value (`DD/MM/YYYY` for the same date with `dayjs`).
+2. It replaces each token with its token from our localization object (i.e: `YYYY` remains `YYYY` for the english locale but becomes `AAAA` for the french locale).
+
+:::warning
+The format returned by `useParsedFormat` cannot be parsed by your date library.
+:::
+
+```js
+import { useParsedFormat } from '@mui/x-date-pickers/hooks';
+
+const parsedFormat = useParsedFormat(props);
+```
+
+##### Updated example
+
+Here is the updated example with a placeholder:
+
+```jsx
+import { useParsedFormat } from '@mui/x-date-pickers/hooks';
+// ... other imports
+
+function CustomDateField(props) {
+  const { format, timezone, value, onChange } = props;
+  const [inputValue, setInputValue] = useInputValue(value, format);
+
+  const placeholder = useParsedFormat(props);
+
+  const { hasValidationError, getValidationErrorForNewValue } = useValidation({
+    value,
+    timezone,
+    props,
+    validator: validateDate,
+  });
+
+  const handleChange = (event) => {
+    const newInputValue = event.target.value;
+    const newValue = dayjs(newInputValue, format);
+    setInputValue(newInputValue);
+    onChange(newValue, { validationError: getValidationErrorForNewValue(newValue) });
+  };
+
+  return (
+    <TextField
+      placeholder={format}
+      value={inputValue}
+      onChange={handleChange}
+      error={hasValidationError}
+    />
+  );
+}
+```
+
+#### Spread props to the DOM
+
+The field receives a lot of props that cannot be forwarded to the DOM element without warnings.
+You can use the `useSplitFieldProps` hook to get the props that can be forwarded safely to the DOM:
+
+```tsx
+const { internalProps, forwardedProps } = useSplitFieldProps(
+  // The props received by the field component
+  props,
+  // The value type ("date", "time" or "date-time")
+  'date',
+);
+
+return (
+  <TextField {...forwardedProps} value={inputValue} onChange={handleChange}>
+)
+```
+
+:::success
+The `forwardedProps` contain props like `slots`, `slotProps` and `sx` that are specific to MUI.
+You can omit them if the component your are forwarding the props to does not support those concepts:
+
+```jsx
+const { slots, slotProps, sx, ...other } = props;
+
+const { internalProps, forwardedProps } = useSplitFieldProps(other, 'date');
+
+return (
+  <TextField {...forwardedProps} value={inputValue} onChange={handleChange}>
```

Comment:
Great addition, but WDYT about using `<input` here to give a bit more precise example (TextField does support those concepts)?
```suggestion
  <input {...forwardedProps} value={inputValue} onChange={handleChange}>
```

Reply:
All the other custom behavior demos are built using MUI.
Until we have Base UI X, I would stick with MUI components here and then do a clean split of what people should build with the MUI X and with the Base UI X packages.

Reply:
Fair point.
We can keep the existing consistency. 👌

Reply:
On the other hand, I stand with my initial proposal.
I think that in this example, to better signal when `sx` and other props won't work, using a non-MUI component in this snippet would be better.

Reply:
I added a 2nd demo with `<input />`, WDYT to keep both? 
Btw #15671 will improve it a lot :laughing:

Reply:
But I was referring only to the callout specifically.
The previous demos made total sense, I just felt as if the callout information should reference `input` instead of `TextField`. 🤔 
WDYT?

Reply:
Ooooooooooh
I'm out of touch today :laughing:

Reply:
Fixed